src/profileflow.cpp

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "leakdetector.h"
+#include "logger.h"
+#include "mozillavpn.h"
+#include "profileflow.h"
+#include "tasks/getsubscriptiondetails/taskgetsubscriptiondetails.h"
+#include "taskscheduler.h"
+
+namespace {
+Logger logger(LOG_MODEL, "ProfileFlow");
+}
+
+ProfileFlow::ProfileFlow() { MVPN_COUNT_CTOR(ProfileFlow); }
+
+ProfileFlow::~ProfileFlow() { MVPN_COUNT_DTOR(ProfileFlow); }
+
+void ProfileFlow::setState(State state) {
+  logger.debug() << "Set state" << state;
+
+  m_state = state;
+  emit stateChanged(m_state);
+}
+
+void ProfileFlow::start() {
+  logger.debug() << "Start profile flow";
+
+  setState(StateLoading);
+
+  User* user = MozillaVPN::instance()->user();
+  Q_ASSERT(user);
+
+  TaskGetSubscriptionDetails* task =
+      new TaskGetSubscriptionDetails(user->email());
+  connect(task, &TaskGetSubscriptionDetails::receivedData, this,
+          &ProfileFlow::subscriptionDetailsFetched);
+  connect(task, &TaskGetSubscriptionDetails::needsAuthentication, this, [&] {
+    logger.debug() << "Needs authentication";
+    setState(StateAuthenticating);
+    emit showProfile();
```

Comment:
I think `setSate` is enough. You should be able to remove `showProfile` entirely.

Reply:
Plus, why showing the profile when you are in `StateAuthenticating` ?